core/ledger/pvtdatastorage/store_created_from_snapshot_test.go

```diff
@@ -0,0 +1,327 @@
+					1: {
+						{
+							Name:             "eligible-coll",
+							MemberOrgsPolicy: iamIn.toMemberOrgPolicy(),
+							BlockToLive:      40,
+						},
+
+						{
+							Name:             "ineligible-coll",
+							MemberOrgsPolicy: iamOut.toMemberOrgPolicy(),
+							BlockToLive:      50,
+						},
+					},
+				},
+			),
+		)
+
+		snapshotDataImporter, err := p.SnapshotDataImporterFor("test-ledger", 25,
+			newMockMembershipProvider("myOrg"), configHistoryMgr.GetRetriever("test-ledger"),
+		)
+		require.NoError(t, err)
+
+		for _, d := range snapshotData {
+			require.NoError(t,
+				snapshotDataImporter.ConsumeSnapshotData("ns", d.collection, d.keyHash, d.valueHash, d.version),
+			)
+		}
+
+		store, err := p.OpenStore("test-ledger")
+		require.NoError(t, err)
+		store.Init(
+			btltestutil.SampleBTLPolicy(
+				map[[2]string]uint64{
+					{"ns", "eligible-coll"}:   40,
+					{"ns", "ineligible-coll"}: 50,
+				},
+			),
+		)
+		return store
+	}
+
+	t.Run("basic-functions-on-store", func(t *testing.T) {
+		store := setup(
+			[]*snapshotData{
+				{
+					collection: "eligible-coll",
+					keyHash:    []byte("eligible-coll-key-hash"),
+					valueHash:  []byte("eligible-coll-value-hash"),
+					version:    version.NewHeight(20, 200),
+				},
+			},
+		)
+
+		require.False(t, store.isEmpty)
+		require.Equal(t, &bootsnapshotInfo{createdFromSnapshot: true, lastBlockInSnapshot: 25}, store.bootsnapshotInfo)
+
+		isEmpty, lastBlkNum, err := store.getLastCommittedBlockNum()
+		require.NoError(t, err)
+		require.False(t, isEmpty)
+		require.Equal(t, uint64(25), lastBlkNum)
+
+		err = store.Commit(25, nil, nil)
+		require.EqualError(t, err, "Expected block number=26, received block number=25")
+		require.NoError(t, store.Commit(26, nil, nil))
+	})
+
+	t.Run("fetch-bootkv-hashes", func(t *testing.T) {
+		store := setup(
+			[]*snapshotData{
+				{
+					collection: "eligible-coll",
+					keyHash:    []byte("eligible-coll-key-hash-1"),
+					valueHash:  []byte("eligible-coll-value-hash-1"),
+					version:    version.NewHeight(20, 200),
+				},
+				{
+					collection: "eligible-coll",
+					keyHash:    []byte("eligible-coll-key-hash-2"),
+					valueHash:  []byte("eligible-coll-value-hash-2"),
+					version:    version.NewHeight(20, 200),
+				},
+			},
+		)
+
+		m, err := store.FetchBootKVHashes(20, 200, "ns", "eligible-coll")
+		require.NoError(t, err)
+		require.Equal(t,
+			map[string][]byte{
+				"eligible-coll-key-hash-1": []byte("eligible-coll-value-hash-1"),
+				"eligible-coll-key-hash-2": []byte("eligible-coll-value-hash-2"),
+			},
+			m,
+		)
+
+		m, err = store.FetchBootKVHashes(25, 200, "ns", "eligible-coll")
+		require.NoError(t, err)
+		require.Len(t, m, 0)
+
+		_, err = store.FetchBootKVHashes(26, 200, "ns", "eligible-coll")
+		require.EqualError(t, err, "unexpected call. Boot KV Hashes are persisted only for the data imported from snapshot")
+	})
+
+	t.Run("committing-old-blocks-pvtdata-deletes-bootKV-hashes", func(t *testing.T) {
```

Comment:
I expected this test to be present in the `reconciler_xxx_test.go`. In general, the test for xxx.go is expected to be found in xxx_test.go. Even certain IDE plugins assume that and there are shortcuts to open an alternative file (test or production code). Breaking this tradition is going to introduce some difficult in the future for sure where we need to search for a particular test as it cannot be found in an alternative file. Hence, I would prefer the test to be present in the alternative file. I leave this to you to decide.